core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java

```diff
@@ -0,0 +1,812 @@
+            for (int n = 0; n < tableNames.size(); n++) {
+                parserCache.get(tableNames.get(n)).close();
+            }
+            parserCache.clear();
+            appendMemory.close();
+            path.close();
+        }
+
+        @Override
+        public boolean run(int workerId) {
+            assert workerId == id;
+            boolean busy = drainQueue();
+            doMaintenance(busy);
+            return busy;
+        }
+
+        private boolean drainQueue() {
+            boolean busy = false;
+            while (true) {
+                try {
+                    long cursor;
+                    while ((cursor = sequence.next()) < 0) {
+                        if (cursor == -1) {
+                            return busy;
+                        }
+                    }
+                    busy = true;
+                    LineTcpMeasurementEvent event = queue.get(cursor);
+                    boolean eventProcessed;
+                    try {
+                        if (event.threadId == id) {
+                            eventProcessed = processNextEvent(event);
+                        } else {
+                            if (event.isRebalanceEvent()) {
+                                eventProcessed = processRebalance(event);
+                            } else {
+                                eventProcessed = true;
+                            }
+                        }
+                    } catch (RuntimeException ex) {
+                        LOG.error().$(ex).$();
+                        eventProcessed = true;
+                    }
+                    if (eventProcessed) {
+                        sequence.done(cursor);
+                    }
+                } catch (RuntimeException ex) {
+                    LOG.error().$(ex).$();
+                }
+            }
+        }
+
+        private void doMaintenance(boolean busy) {
+            long millis = milliClock.getTicks();
+            if (busy && (millis - lastMaintenanceJobMillis) < maintenanceJobHysteresisInMs) {
+                return;
+            }
+
+            lastMaintenanceJobMillis = millis;
+            ObjList<CharSequence> tableNames = parserCache.keys();
+            for (int n = 0, sz = tableNames.size(); n < sz; n++) {
+                Parser parser = parserCache.get(tableNames.get(n));
+                parser.doMaintenance();
+            }
+        }
+
+        private boolean processNextEvent(LineTcpMeasurementEvent event) {
+            Parser parser = parserCache.get(event.getTableName());
+            if (null == parser) {
+                parser = new Parser();
+                try {
+                    parser.processFirstEvent(engine, securityContext, event);
+                } catch (CairoException ex) {
+                    LOG.info().$(name).$(" could not create parser [name=").$(event.getTableName()).$(", ex=").$(ex.getFlyweightMessage()).$(']').$();
```

Comment:
This never happens in tests. We should also catch "CairoError". writer.commit() is liable to throw that.

Could we stick to "message [context]" style of logging for the thread name too?